Frontend/src/components/playlistComponents/videoMetaforPlaylist.jsx

```diff
@@ -0,0 +1,201 @@
+import React, { useContext, useEffect, useRef, useState } from "react";
+import axios from "axios";
+import { PlaylistContext } from "../UseContext/PlaylistContext";
+import GetTimeStamp from "./getTimeStamp";
+
+function VideoMetaforPlaylist() {
+  const [updatedPlaylist, setupdatedPlaylist] = useState('')
+  const {
+    videoLinkAdded,
+    playListData,
+    setplayListData,
+    playListId,
+    playListStack
+  } = useContext(PlaylistContext);
+
+  const Api_key = "AIzaSyCLQGxPXu7sr-mFVgoHHQKU8EcduZHbeFM";
+  const url = videoLinkAdded;
+  const videoId = url.split("v=")[1];
+
+  const prevVideoIdRef = useRef(null);
+
+  const [isLoading, setIsLoading] = useState(false); // Add loading state
+
+  // ...
+  
+  useEffect(() => {
+    if (videoId && prevVideoIdRef.current !== videoId) {
+      setIsLoading(true); // Start loading
+  
+      axios
+        .get(
+          `https://www.googleapis.com/youtube/v3/videos?part=snippet,contentDetails&id=${videoId}&key=${Api_key}`
+        )
+        .then((res) => {
+          const newVideo = {
+            link: videoLinkAdded,
+            title: res.data.items[0].snippet.title,
+            thumbnail: res.data.items[0].snippet.thumbnails.default.url,
+            duration: res.data.items[0].contentDetails.duration,
+          };
+  
+          setplayListData((prev) => ({
+            ...prev,
+            videos: prev.videos
+              ? [
+                  ...prev.videos,
+                  {
+                    id: prev.videos.length,
+                    videoMetaData: { ...newVideo },
+                  },
+                ]
+              : [
+                  {
+                    id: 0,
+                    videoMetaData: { ...newVideo },
+                  },
+                ],
+          }));
+  
+          setIsLoading(false); // Stop loading
+          console.log(videoId)
+        })
+        .catch((error) => {
+          console.log(error);
+          setIsLoading(false); // Stop loading on error
+        });
+  
+      prevVideoIdRef.current = videoId;
+    }
+  }, [videoId]);
+  return (
+    // <div>
```

Comment:
Remove the commented code from PR, you can refer to previous commits if need to refer code later.